### ISSUE TITLE
Deprecate Docker Resource Type

### ISSUE THREAD

**Issue by scottietremendous** [concourse/concourse#6586]
### TLDR

- Deprecate the `docker-image-resource` in favour of the oci-build prototype

### Problem

- Currently, Concourse utilizes the [`docker-image resource`](https://github.com/concourse/docker-image-resource) to support checking, fetching, and pushing images to Docker registries. 
- Related to the problems outlined in #6584 the push/pull process for image building can be quite clunky and verbose using the current [`oci-build-task`](https://hub.docker.com/r/vito/oci-build-task)

### Proposed Solution

- Utilize Prototypes instead! 
- Deprecate the `docker-image-resource` and utilize the proposed oci-build prototype instead

### Outcomes

- OCI images become easier to build and push
- Users are provided the same complexity but in an easier to manage format 
- Provide a better UX for building images

### Current Status

- Blocked by ongoing initial prototypes work

### References

https://github.com/concourse/concourse/pull/6540 - Add support for prototypes
[Replace Docker Image Resource Type](https://github.com/concourse/registry-image-resource/milestone/8)

See: [Reinventing Resource Types](https://blog.concourse-ci.org/reinventing-resource-types/)

**Comment by aoldershaw** [comment]
The references to registry-image resource should be docker-image resource if I'm not mistaken 😀

**Comment by scottietremendous** [comment]
@aoldershaw - Right you are! Had resource types dancing through my head.

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
